addons/l10n_vn_edi_viettel/tests/test_edi.py

```diff
@@ -0,0 +1,347 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import datetime
+from unittest import mock
+from unittest.mock import patch
+
+from freezegun import freeze_time
+
+from odoo import fields
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+from odoo.tests import tagged
+
+
+@tagged('post_install_l10n', 'post_install', '-at_install')
+class TestVNEDI(AccountTestInvoicingCommon):
+
+    @classmethod
+    def setUpClass(cls, chart_template_ref='vn'):
+        super().setUpClass(chart_template_ref=chart_template_ref)
+
+        # Setup the default symbol and template.
+        cls.template = cls.env['l10n_vn_edi_viettel.sinvoice.template'].create({
+            'name': '1/001',
+            'template_invoice_type': '1',
+        })
+        cls.symbol = cls.env['l10n_vn_edi_viettel.sinvoice.symbol'].create({
+            'name': 'K24TUT',
+            'invoice_template_id': cls.template.id,
+        })
+        cls.env['ir.property']._set_default(
+            'l10n_vn_edi_symbol',
+            'res.partner',
+            cls.symbol,
+            cls.env.company.id
+        )
+
+        # Setup a vietnamese address on the partner and company.
+        cls.partner_a.write({
+            'street': '121 Hang Bac Street',
+            'state_id': cls.env.ref('base.state_vn_VN-HN').id,
+            'city': 'Hoan Kiem District',
+            'country_id': cls.env.ref('base.vn').id,
+            'vat': '0100109106-505',
+            'phone': '3825 7670',
+            'email': 'partner_a@gmail.com',
+        })
+
+        cls.env.company.write({
+            'street': '3 Alley 45 Phan Dinh Phung, Quan Thanh Ward',
+            'state_id': cls.env.ref('base.state_vn_VN-HN').id,
+            'country_id': cls.env.ref('base.vn').id,
+            'vat': '0100109106-506',
+            'phone': '6266 1275',
+            'email': 'test_company@gmail.com',
+            'website': 'test_company.com',
+            'l10n_vn_edi_password': 'a',
+            'l10n_vn_edi_username': 'b',
+        })
+
+        cls.product_a.default_code = 'BN/1035'
+
+        cls.fakenow = datetime(2024, 1, 1)
+        cls.startClassPatcher(freeze_time(cls.fakenow))
```

Comment:
I prefer an explicit freeze_time on test methods. It bring more flexibility, specially when you have to deal with multi-currency at != dates, payments etc